simulator.go

```diff
@@ -203,6 +204,8 @@ type simulatorAPIHandler struct {
 	nodes  map[string]*docker.Container
 
 	result *simulationResult
+	nodesMutex sync.RWMutex
```

Comment:
Please only call it lock, it will be needed when updating subresults too concurrently.